hazelcast/src/main/java/com/hazelcast/map/impl/record/Records.java

```diff
@@ -29,37 +34,33 @@
     private Records() {
     }
 
-    public static void applyRecordInfo(Record record, RecordInfo replicationInfo) {
-        record.setVersion(replicationInfo.getVersion());
-        record.setHits(replicationInfo.getHits());
-        record.setTtl(replicationInfo.getTtl());
-        record.setMaxIdle(replicationInfo.getMaxIdle());
-        record.setCreationTime(replicationInfo.getCreationTime());
-        record.setLastAccessTime(replicationInfo.getLastAccessTime());
-        record.setLastUpdateTime(replicationInfo.getLastUpdateTime());
-        record.setExpirationTime(replicationInfo.getExpirationTime());
-        record.setLastStoredTime(replicationInfo.getLastStoredTime());
+    public static void writeRecord(ObjectDataOutput out, Record record, Data dataValue) throws IOException {
+        getById(record.getMatchingRecordReaderWriterId()).writeRecord(out, record, dataValue);
```

Comment:
Maybe write the `record.getMatchingRecordReaderWriterId()` here, just as it is read in `Records#readRecord`.